dev/release/cherry-pick-pr.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/python3
```

Comment:
This script is used to do maintenance releases (e.g. 5.1.0, 5.2.0, etc)

It might be worth considering if we want to do maintenance releases of datafusion at all, at least initially. It takes me non trivial effort to do this for arrow, but I think there are enough downstream projects (other than datafusion) that I felt it was worth the investment in time for arrow-rs. 

I am not sure about datafusion-rs if most downstream projects pull directly from git (I don't know if this is true or not)

Reply:
I agree, I removed the cherry-pick automation. Let's start with something simple and iterate from there. Git dependency is usually not a problem for projects that have executables as release artifacts. For libraries like delta-rs, using git reference to pin the dependency doesn't work very well because it prevents it from being able to be published to crates.io. But I think we can revisit this when another 5.x patch or minor release is needed.